### PR TITLE
ROMIO: add --enable-fortran configure option

### PR DESCRIPTION
**Pull request by wkliao**
## Pull Request Description
Note MPICH configure option --enable-fortran/--disable-fortran is
passed down to ROMIO.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by hzhou** [comment]
@wkliao This one looks fine to merge. Could you rebase this PR?

**Review by hzhou — CHANGES_REQUESTED** [review]
Sorry that I forgot about this one. The options in main mpich configure has changed a little. The current fortran options are -- https://github.com/pmodels/mpich/blob/3494619ecdd413427f7638e274c9f70a1960e80a/configure.ac#L441-L452

**Comment by wkliao** [comment]
> Sorry that I forgot about this one. The options in main mpich configure has changed a little. The current fortran options are --
> 
> https://github.com/pmodels/mpich/blob/3494619ecdd413427f7638e274c9f70a1960e80a/configure.ac#L441-L452

These are already in my changes.

File: src/mpi/romio/configure.ac [hunk 1]
```diff
@@ -241,11 +237,35 @@ dnl
...
+                    [Turn off support for Fortran. @<:@default: enabled@:>@])],
+    [enable_fortran=${enableval}], [enable_fortran=yes])
+AC_ARG_ENABLE(f77,
+    [AS_HELP_STRING([--disable-f77],
+                    [Turn off support for Fortran 77. @<:@default: enabled@:>@])],
+    [enable_f77=${enableval}], [enable_f77=yes])
+AC_ARG_ENABLE(fc,
+    [AS_HELP_STRING([--disable-fc],
+                    [Turn off support for Fortran 90 and 2008. @<:@default: enabled@:>@])],
+    [enable_fc=${enableval}], [enable_fc=yes])
```
Comment (hzhou) [inline src/mpi/romio/configure.ac:252]: MPICH now uses `--disable-f90` rather than than `--disable-fc`.

File: src/mpi/romio/configure.ac [hunk 2]
```diff
@@ -587,11 +619,11 @@ if test -n "$OFFSET_KIND" -a "A$MPI_OFFSET_KIND1" = "A!" ; then
   MPI_OFFSET_KIND2="        PARAMETER (MPI_OFFSET_KIND=$OFFSET_KIND)"
   MPI_OFFSET_KIND_VAL=$OFFSET_KIND
 else
- if test "$FORTRAN_MPI_OFFSET" = "integer*8" && test "A$MPI_OFFSET_KIND2" = "A!" && test $NOF77 = 0 && test $NOF90 = 0 ; then
+ if test "$FORTRAN_MPI_OFFSET" = "integer*8" && test "A$MPI_OFFSET_KIND2" = "A!" && test $NOF77 = 0 && (test $NOF90 = 0 || test $NOF08 = 0) ; then
```
Comment (hzhou) [inline src/mpi/romio/configure.ac:622]: I am not quite clear about the logic here. Can someone explain in plain English? @wkliao @roblatham00

**Comment by wkliao** [comment]
That is a very old code. I believe it is to figure out MPI_OFFSET_KIND and save in include/mpiof.h, as KIND is introduced in F90.
```
      INTEGER MPI_OFFSET_KIND
      PARAMETER (MPI_OFFSET_KIND=           8)
```
or
```
      INTEGER MPI_OFFSET_KIND
      PARAMETER (MPI_OFFSET_KIND=           4)
```

**Comment by hzhou** [comment]
> That is a very old code. I believe it is to figure out MPI_OFFSET_KIND and save in include/mpiof.h, as KIND is introduced in F90.
> 
> ```
>       INTEGER MPI_OFFSET_KIND
>       PARAMETER (MPI_OFFSET_KIND=           8)
> ```
> 
> or
> 
> ```
>       INTEGER MPI_OFFSET_KIND
>       PARAMETER (MPI_OFFSET_KIND=           4)
> ```

If it is trying to test whether it is F90 compiler, I think we can simply remove the check. MPICH is now requiring F90 compilers. And today it is difficult to find a legacy F77 compiler in the wild anyway.

**Comment by wkliao** [comment]
It is not to test whether the Fortran compiler is a F90.
That codes update include/mpiof.h.
However, I think mpiof.h can be removed completely.
as mpif.h has already covered all the necessities.
There can be a separate new PR for that task.
FYI. A related PR is #4693

**Comment by hzhou** [comment]
> It is not to test whether the Fortran compiler is a F90. That codes update include/mpiof.h. However, I think mpiof.h can be removed completely. as mpif.h has already covered all the necessities. There can be a separate new PR for that task. FYI. A related PR is #4693

#4693 is merged!

**Comment by wkliao** [comment]
Close as this affects only #6956
